Source/playerinfo/PlayerInfo.cpp

```diff
@@ -0,0 +1,406 @@
+                        newArray[numberOfCodecs] = PLAYERINFO_AUDIO_WAV;
+                        break;
+                    }
+                    ++numberOfCodecs;
+                }
+                if (numberOfCodecs < length) {
+                    value = numberOfCodecs;
+                    ::memcpy(array, newArray, numberOfCodecs * sizeof(playerinfo_audiocodec_t));
+                } else {
+                    value = -numberOfCodecs;
+                }
+            }
+        }
+
+        return value;
+    }
+};
+
+/* static */ PlayerInfo::PlayerInfoAdministration PlayerInfo::_administration;
+} //namespace WPEFramework
+
+using namespace WPEFramework;
+extern "C" {
+
+struct playerinfo_type* playerinfo_instance(const char name[])
+{
+    if (name != NULL) {
+        return reinterpret_cast<playerinfo_type*>(PlayerInfo::Instance(string(name)));
+    }
+    return NULL;
+}
+
+void playerinfo_release(struct playerinfo_type* instance)
+{
+    if (instance != NULL) {
+        reinterpret_cast<PlayerInfo*>(instance)->Release();
+    }
+}
+
+playerinfo_playback_resolution_t playerinfo_playback_resolution(struct playerinfo_type* instance)
+{
+
+    playerinfo_playback_resolution_t result = PLAYERINFO_RESOLUTION_UNKNOWN;
+    if (instance != NULL) {
+        switch (reinterpret_cast<PlayerInfo*>(instance)->PlaybackResolution()) {
+        case Exchange::IPlayerProperties::PlaybackResolution::RESOLUTION_UNKNOWN:
+            result = PLAYERINFO_RESOLUTION_UNKNOWN;
+            break;
+        case Exchange::IPlayerProperties::PlaybackResolution::RESOLUTION_480I:
+            result = PLAYERINFO_RESOLUTION_480I;
+            break;
+        case Exchange::IPlayerProperties::PlaybackResolution::RESOLUTION_480P:
+            result = PLAYERINFO_RESOLUTION_480P;
+            break;
+        case Exchange::IPlayerProperties::PlaybackResolution::RESOLUTION_576I:
+            result = PLAYERINFO_RESOLUTION_576I;
+            break;
+        case Exchange::IPlayerProperties::PlaybackResolution::RESOLUTION_576P:
+            result = PLAYERINFO_RESOLUTION_576P;
+            break;
+        case Exchange::IPlayerProperties::PlaybackResolution::RESOLUTION_720P:
+            result = PLAYERINFO_RESOLUTION_720P;
+            break;
+        case Exchange::IPlayerProperties::PlaybackResolution::RESOLUTION_1080I:
+            result = PLAYERINFO_RESOLUTION_1080I;
+            break;
+        case Exchange::IPlayerProperties::PlaybackResolution::RESOLUTION_1080P:
+            result = PLAYERINFO_RESOLUTION_1080P;
+            break;
+        case Exchange::IPlayerProperties::PlaybackResolution::RESOLUTION_2160P30:
+            result = PLAYERINFO_RESOLUTION_2160P30;
+            break;
+        case Exchange::IPlayerProperties::PlaybackResolution::RESOLUTION_2160P60:
+            result = PLAYERINFO_RESOLUTION_2160P60;
+            break;
+        default:
+            result = PLAYERINFO_RESOLUTION_UNKNOWN;
```

Comment:
Can we have an error message here to give a hint that the enums here and in the interface do not match anymore?